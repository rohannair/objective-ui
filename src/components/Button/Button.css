@import '../../styles/globals';
@define-extend regularbutton {
  background-color: #fff;
  display: flex;
  overflow: hidden;
  position: relative;
  color: rgba(0, 0, 0, 0.87);
  font-size: 0.875rem;
  letter-spacing: 0.03em;
  line-height: 2.5;
  text-transform: uppercase;
  transition: all 0.2s ease-in-out;
  height: 36px;
  padding: 0 20px;
  border: none;
  border-radius: 36px;
  cursor: pointer;
  outline: none;
  justify-content: center;
  vertical-align: middle;
  white-space: nowrap;
  user-select: none;
}

.primary {
  @extend regularbutton;

  background-color: $primary;
  color: $background;

  &:hover {
    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.2);
    color: #fff;
  }
}

.secondary {
  @extend regularbutton;

  background-color: $secondary;
  color: $background;

  &:hover {
    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.2);
    color: #fff;
  }
}

.transparent {
  @extend regularbutton;

  background-color: transparent;
  border: 2px dashed rgba(0, 0, 0, 0.2);
}

.link {
  @extend regularbutton;

  background-color: transparent;
  color: $secondary;
  border: 1px solid rgba(0, 0, 0, 0.2);
}

.sm {
  padding: 0 12px;
}

.right {
  margin-left: auto;
}
