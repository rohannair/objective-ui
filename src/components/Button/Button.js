import React, { PropTypes } from 'react';
import styles from './Button.css';
import classNames from 'classnames/bind';

let cx = classNames.bind(styles);

const Button = p => {
  const className = cx({
    [styles.primary]: p.primary,
    [styles.secondary]: p.secondary,
    [styles.transparent]: p.transparent,
    [styles.sm]: p.size === 'sm',
    [styles.right]: p.right,
    [styles.link]: p.link || (!p.primary && !p.secondary)
  });

  return (
    <button className={className} onClick={p.onClick}>
      { p.children }
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element
  ])
};

export default Button;
