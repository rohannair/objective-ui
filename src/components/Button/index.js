import Button, { StyledButton as sb } from './Button'
export default Button
export const StyledButton = sb
