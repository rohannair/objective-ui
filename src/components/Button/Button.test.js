import React from 'react'
import Button from './index.js'

describe('Test suite for Button component', () => {
  it('Button should exist', () => {
    let wrapper = shallow(<Button/>)
    expect(wrapper).toMatchSnapshot()
  })
})

