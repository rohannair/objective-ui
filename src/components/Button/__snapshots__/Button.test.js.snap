exports[`Test suite for Button component Button should exist 1`] = `
<div
  className="" />
`;
